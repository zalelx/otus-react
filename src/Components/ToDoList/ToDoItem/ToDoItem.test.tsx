import renderer from "react-test-renderer";
import { ToDoItem } from "./index";
import React from "react";
import { shallowToJson } from "enzyme-to-json";
import { shallow } from "enzyme";

const element = (
  <ToDoItem checked onCheck={jest.fn()} onRemove={jest.fn()} task="abc" />
);

describe("render ToDoItem with different renderers", () => {
  it("with react renderer", () => {
    const newVar = renderer.create(element).toJSON();

    expect(newVar).toMatchSnapshot();
  });

  it("with enzyme renderer", () => {
    const json = shallowToJson(shallow(element));

    expect(json).toMatchSnapshot();
  });
});
