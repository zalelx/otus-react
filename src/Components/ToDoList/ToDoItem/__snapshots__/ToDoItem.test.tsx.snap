// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ToDoItem with different renderers with enzyme renderer 1`] = `
<div>
  "abc"
  <button
    onClick={[MockFunction]}
  >
    X
  </button>
</div>
`;

exports[`render ToDoItem with different renderers with react renderer 1`] = `
.emotion-0 {
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

<div>
  <span
    className="emotion-0"
    onClick={[MockFunction]}
  >
    abc
  </span>
  <button
    onClick={[MockFunction]}
  >
    X
  </button>
</div>
`;
